hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java

```diff
@@ -294,6 +294,14 @@ public void pause() {
     getLifeCycle().transition(LifeCycle.State.PAUSED);
   }
 
+  @Override
+  public void reinitialize() {
+    if (getLifeCycleState() == LifeCycle.State.PAUSED) {
```

Comment:
cc @hanishakoneru does OmStateMachine also need similar code?

Reply:
Question: How these tests are working before? (Not able to understand that? can you shed some info)

Reply:
After `notifyInstallSnapshotFromLeader` is called, ratis calls 
```
                if (reply != null) {
                  LOG.info("{}: StateMachine successfully installed snapshot index {}. Reloading the StateMachine.",
                      getMemberId(), reply.getIndex());
                  stateMachine.pause();
                  state.updateInstalledSnapshotIndex(reply);
                  state.reloadStateMachine(reply.getIndex());
                }
```
`stateMachine.pause();` will make SM to be in PAUSED state, `state.reloadStateMachine(reply.getIndex())` will trigger `StateMachineUpdater#reload()` to be called, which will then call `stateMachine.reinitialize();`.

This is the reason of the fix.

As far as I known,  at the end of `TestSCMInstallSnapshotWithHA#testInstallSnapshot`, the followerSCM is also in PAUSED state, which is not checked before.

And for `testInstallOldCheckpointFailure`, `notifyInstallSnapshotFromLeader` is not really called, since without the fix in RATIS-1369, downloading snapshot taken at index 0 is ignore by follower SCM.